### ISSUE TITLE
Plugin editor broken/freezing in IntelliJ 2019.1 EAP w/ JRE11 due to JavaFX errors 

### ISSUE THREAD

**Issue by bentolor** [asciidoctor/asciidoctor-intellij-plugin#238]
IntelliJ 2019.1 will (optionally?) come with a patched JRE11. On editing any Asciidoc document the Asciidoctor plugin intermittently slightly freezes and hangs spilling out the quoted errors below.

You can comfortably download & try the EAP builds using the IntelliJ Toolbox application: 

I assume this is some incompability due to JavaFX in JDK8 vs. JDK11. There it is the "IntelliJ + JBR11" package.

````
java.lang.NoSuchMethodError: javafx.scene.Scene.impl_getPeer()Lcom/sun/javafx/tk/TKScene;
	at org.asciidoc.intellij.editor.javafx.JFXPanelWrapper.addNotify(JFXPanelWrapper.java:45)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at com.intellij.ui.JBSplitter.addNotify(JBSplitter.java:109)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addImpl(Container.java:1146)
	at com.intellij.ui.tabs.impl.JBTabsImpl.addImpl(JBTabsImpl.java:2654)
	at java.desktop/java.awt.Container.add(Container.java:436)
	at com.intellij.ui.tabs.impl.JBTabsImpl.updateContainer(JBTabsImpl.java:2628)
	at com.intellij.ui.tabs.impl.JBTabsImpl.updateAll(JBTabsImpl.java:821)
	at com.intellij.ui.tabs.impl.JBTabsImpl.processRemove(JBTabsImpl.java:2547)
	at com.intellij.ui.tabs.impl.JBTabsImpl.removeTab(JBTabsImpl.java:2505)
	at com.intellij.ui.tabs.impl.JBTabsImpl.removeTab(JBTabsImpl.java:2476)
	at com.intellij.openapi.fileEditor.impl.EditorTabbedContainer.removeTabAt(EditorTabbedContainer.java:280)
	at com.intellij.openapi.fileEditor.impl.EditorWindow.lambda$closeFile$1(EditorWindow.java:234)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.runChange(FileEditorManagerImpl.java:1672)
	at com.intellij.openapi.fileEditor.impl.EditorWindow.closeFile(EditorWindow.java:214)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$closeFile$1(FileEditorManagerImpl.java:609)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:220)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:178)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:168)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:154)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.closeFile(FileEditorManagerImpl.java:607)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.closeFile(FileEditorManagerImpl.java:617)
	at com.intellij.openapi.fileEditor.impl.EditorTabbedContainer$CloseTab.actionPerformed(EditorTabbedContainer.java:477)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:260)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:277)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:624)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$processAction$2(IdeKeyEventDispatcher.java:673)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:672)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.java:520)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:475)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:217)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:647)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:367)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
````

AsciiDoc Plugin Version: v0.24.1

IDE Name and Version: IntelliJ IDEA 2019.1 EAP (Ultimate Edition)
Build #IU-191.5701.16, built on February 20, 2019
IntelliJ IDEA EAP User
Expiration date: March 22, 2019
JRE: 11-ea+96 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.18.0-15-generic

**Comment by ahus1** [comment]
Thanks for reporting this. Looking at the code this only happens in HiDPI mode. I will need to do some preparations to reproduce it here.

**Comment by bentolor** [comment]
@ahus1 Interestingly: so my FullHD 1920x1080 14" Laptop screen counts as HiDPI. Uh-hu…

**Comment by ahus1** [comment]
As far as I know hidpi starts when you set your font size to something bigger than 100%. I tried to reproduce it, but didn't manage to do so. 

I've added a fix and released and published a pre-released 0.24.2. As I wasn't able to reproduce it, I was also unable to test if this fix works. I would like you to give it a try.

You can download it from the [GitHub releases plugin](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases).

Please comment in this ticket if this resolves this issue. Thanks!

**Comment by bentolor** [comment]
Thanks for your quick response, Alexander. But I have some bad news. 

This change seems to have made it even worse. Now, on opening the project with the asciidoctor plugin enabled, the JVM core dumps.

My only rescue was to delete the asciidoctor plugin-directory, which resolves this issues.

The stack clearly points towards JavaFX…

See on core dump here: https://gist.github.com/bentolor/763c2c73b70b4b3ce905f33c147808eb

**Comment by ahus1** [comment]
Thanks for reporting this. The top of the stack trace is "URLLoader.twkDidReceiveResponse" - did your Asciidoc documents reference an external image? Does this happen as well when showing an empty document?

Using Google there seemed to be similar reports of this problem, but they are apparently hard to reproduce. I suggest you use the URL at the top of the heap dump: http://bugreport.java.com/bugreport/crash.jsp

If you want to use JDK11, you can disable JavaFX and use the Swing preview (see the plugin's settings for this). I know this is a drawback, but maybe the only workaround I currently have for you.

**Comment by ahus1** [comment]
... side note: you might alternatively report this to JetBrains, as they might have packaged/patched the JDK including the included JavaFX.

**Comment by bentolor** [comment]
Hi @ahus1 

1. No - the document does not refer to any external images
2. The crash is stable reproducible by creating a new file `test.adoc` : Immediate SIGSEVs
3. I miss time, so I'm unable to verify if the crash is specific to JBR or upstream JVM
4. I reported https://youtrack.jetbrains.com/issue/JBR-1256

**Comment by bentolor** [comment]
Can confirm: plain 2019.1 still works!

**Comment by gisthere** [comment]
I have similar issue after installing asciidoc plugin:
```
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.ide.ClassUtilCore to field sun.net.www.protocol.jar.JarFileFactory.fileCache
WARNING: Please consider reporting this to the maintainers of com.intellij.ide.ClassUtilCore
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f2393ba2a99, pid=7921, tid=7975
#
# JRE version: OpenJDK Runtime Environment (11.0.2+9) (build 11.0.2+9-b159.30)
# Java VM: OpenJDK 64-Bit Server VM (11.0.2+9-b159.30, mixed mode, tiered, compressed oops, concurrent mark sweep gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x8c1a99]  jni_CallStaticObjectMethodV+0x79
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P" (or dumping to /home/fanis/Downloads/pycharm-2019.1/core.7921)
#
# An error report file with more information is saved as:
# /home/user/java_error_in_PYCHARM_7921.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
Aborted (core dumped)

```

**Comment by bentolor** [comment]
@gisthere Yep: You seem to run into the same bug. 

`# V [libjvm.so+0x8c1a99] jni_CallStaticObjectMethodV+0x79`

So best chip in in the linked Jetbrains issue. Alexander can't do much about this. This bug needs to be fixed upstream. Alternatively you can to switch to JDK8 or Swing-based rendering.

**Comment by ahus1** [comment]
The upcoming release of IntelliJ 2019.2 will come as default with JDK11. They have did some upgrades to the JDK (so latest fixes would be included). 

I still can't reproduce the issue. Therefore I ask those watching this issue: Could you please try again with either the 2019.1.3 or 2019.2 EAP if this issue still exists?

Thanks!

**Comment by unlimitedsola** [comment]
just tried 2019.2 EAP, haven't encountered any crashing after few hours of use.

**Comment by bentolor** [comment]
Just tested with IDEA 2019.1.3 with JBR and seems to work now without any crashes. 

So issue seems to be fixed in IDEA 2019.1.3+ and IDEA 2019.2+

Closing my issue.
